### PR TITLE
[PM-9884] Create empty send add/edit page for browser

### PR DESCRIPTION
**Pull request by djsmith85**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-9884

## 📔 Objective

- Introduces conditional routing based on extension refresh feature flag
- Uses the SendForm introduced with #10147 
- After selecting a Send type via the New button navigate to the new send-add-edit page and build a SendFormConfig

Stacked on top of #10147 and can be merged into it once approved. Tests will occur on main as all functionality is guarded behind a feature flag

## 📸 Screenshots

https://github.com/user-attachments/assets/3e70dab3-bac3-413e-b151-ae68616e0a53

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts
Comment (audreyality) [inline apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts:1]: ⚠️ The component is net-new code and should be unit tested.

File: apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import { PopupHeaderComponent } from "../../../../platform/popup/layout/popup-header.component";
+import { PopupPageComponent } from "../../../../platform/popup/layout/popup-page.component";
+
+/**
+ * Helper class to parse query parameters for the AddEdit route.
+ */
+class QueryParams {
+  constructor(params: Params) {
+    this.sendId = params.sendId;
+    this.type = parseInt(params.type, null);
```
Comment (audreyality) [inline apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts:31]: ⛏️ I think the radix parameter should be 10? `null` is the same as not specifying the parameter, which causes `parseInt` to infer a value when it parses `params.type`.

File: apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            mode = "add";
+          } else {
+            mode = "edit";
+          }
+          const config = await this.addEditFormConfigService.buildConfig(
+            mode,
+            params.sendId,
+            params.type,
+          );
+          return config;
```
Comment (audreyality) [inline apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts:107]: 🤔 This code seems really out of place, but I'm struggling to figure out why. It just seems smelly. 

* Configuration should consist only of metadata, but this method [loads a send](https://github.com/bitwarden/clients/blob/f5eec3d8807ea8f05f71459bd249a5efd4d4ad2e/libs/tools/send/send-ui/src/send-form/services/default-send-form-config.service.ts#L29-L31).
* `switchMap` completely replaces the observable, but the code you're using is performing an async map. Usually you want to use `mergeMap` in that situation.
* It's odd that the service is not only so specialized, but that it has only one public method. That usually indicates that code has been too decomposed, unless there's somewhere you're going with it.

File: apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        }),
+      )
+      .subscribe((config) => {
+        this.config = config;
+        this.headerText = this.setHeader(config.mode, config.sendType);
+      });
+  }
+
+  setHeader(mode: SendFormMode, type: SendType) {
+    const partOne = mode === "edit" || mode === "partial-edit" ? "editItemHeader" : "newItemHeader";
```
Comment (audreyality) [inline apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts:117]: 🎨 A better name for this variable is `headerKey`

File: libs/tools/send/send-ui/src/send-form/send-form.stories.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  },
+};
+
+export const Edit: Story = {
+  ...Default,
+  args: {
+    config: {
+      ...defaultConfig,
+      mode: "edit",
+      originalSend: defaultConfig.originalSend,
```
Comment (vleague2) [inline libs/tools/send/send-ui/src/send-form/send-form.stories.ts:106]: Why do we need to specify the `originalSend` again when it's already in the `defaultConfig` that is being spread?

File: libs/tools/send/send-ui/src/send-form/send-form.stories.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        {
+          provide: ToastService,
+          useValue: {
+            showToast: action("showToast"),
+          },
+        },
+      ],
+    }),
+    componentWrapperDecorator(
+      (story) => `<div class="tw-bg-background-alt tw-text-main tw-border">${story}</div>`,
```
Comment (vleague2) [inline libs/tools/send/send-ui/src/send-form/send-form.stories.ts:67]: If you want to show a border, you also need to add `tw-border-solid` in addition to the `tw-border` you have.

**Review by danielleflinn — DISMISSED** [review]

File: apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+class QueryParams {
+  constructor(params: Params) {
+    this.sendId = params.sendId;
+    this.type = parseInt(params.type, null);
+  }
+
+  /**
+   * The ID of the send to edit
+   */
+  sendId?: SendId;
```
[the remark this answers is not in the document]
Comment (djsmith85) [reply apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts]: Implemented with 2954f8fc22f038c5db0a95a33943edf42e6bc23a

File: apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    FormsModule,
+    ButtonModule,
+    PopupPageComponent,
+    PopupHeaderComponent,
+    PopupFooterComponent,
+    SendFormModule,
+    AsyncActionsModule,
+  ],
+})
+export class SendAddEditComponent {
```
[the remark this answers is not in the document]
Comment (djsmith85) [reply apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts:68]: Implemented with 2954f8fc22f038c5db0a95a33943edf42e6bc23a

File: apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts (hunk 3 above)
Comment (djsmith85) [reply apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts]: Added with 80ffee207676ee5d164df10db53560d415aa4cc8

File: apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+  setHeader(mode: SendFormMode, type: SendType) {
+    const partOne = mode === "edit" || mode === "partial-edit" ? "editItemHeader" : "newItemHeader";
+
+    switch (type) {
+      case SendType.Text:
+        return this.i18nService.t(partOne, this.i18nService.t("sendTypeText"));
+      case SendType.File:
+        return this.i18nService.t(partOne, this.i18nService.t("sendTypeFile"));
+    }
+  }
```
[the remark this answers is not in the document]
Comment (djsmith85) [reply apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts]: Implemented with 80ffee207676ee5d164df10db53560d415aa4cc8

File: apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts (hunk 1 above)
Comment (djsmith85) [reply apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts]: Implemented with 23617d1ed27304270f2c93be7934c7a043b965df

File: apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts (hunk 3 above)
Comment (audreyality) [reply apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.ts]: That commit doesn't change the name of `partOne`.

**Review by audreyality — APPROVED** [review]

File: apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.html [hunk 9]
```diff
@@ -0,0 +1,17 @@
+<popup-page>
```
Comment (vleague2) [inline apps/browser/src/tools/popup/send-v2/add-edit/send-add-edit.component.html:1]: 💭 I'm not a codeowner on this PR anymore, but if you did still want to implement the built-in loading state, you'd pass `[loading]="whatever"` here on `popup-page`.

**Pull request merged**